packages/core/src/modules/oob/messages/OutOfBandInvitation.ts

```diff
@@ -71,17 +73,65 @@ export class OutOfBandInvitation extends AgentMessage {
   public static async fromUrl(invitationUrl: string) {
     const parsedUrl = parseUrl(invitationUrl).query
     const encodedInvitation = parsedUrl['oob']
-
-    if (typeof encodedInvitation === 'string') {
-      const invitationJson = JsonEncoder.fromBase64(encodedInvitation)
-      const invitation = this.fromJson(invitationJson)
-
-      return invitation
-    } else {
-      throw new AriesFrameworkError(
-        'InvitationUrl is invalid. It needs to contain one, and only one, of the following parameters; `oob`'
-      )
+    try {
+      if (typeof encodedInvitation === 'string') {
+        const invitationJson = JsonEncoder.fromBase64(encodedInvitation)
+        const invitation = this.fromJson(invitationJson)
+
+        return invitation
+      } else {
+        throw new AriesFrameworkError(
+          'InvitationUrl is invalid. It needs to contain one, and only one, of the following parameters; `oob`'
+        )
+      }
+    } catch (error) {
+      return await this.fromShortUrl(invitationUrl)
+    }
+  }
+  //This currently does not follow the RFC because of issues with fetch, currently uses a janky work around
+  public static async fromShortUrl(invitationUrl: string) {
```

Comment:
This seems almost identical to the connection fromShortUrl, which makes sense, but is there are way to unify them so we do not repeat themselves.

Reply:
These have been combined into one function